jscomp/bsb/bsb_ninja_regen.ml

```diff
@@ -53,26 +60,32 @@ let regenerate_ninja
   | Other _ -> 
     if check_result = Bsb_bsc_version_mismatch then begin 
       Bsb_log.warn "@{<info>Different compiler version@}: clean current repo@.";
-      Bsb_clean.clean_self  per_proj_dir; 
-    end ; 
+      Bsb_clean.clean_self  build_artifacts_dir; 
```

Comment:
Actually since you're passing the right path here, you don't need to call the `get_build_artifacts_location` inside of it.